example/hook.go

```diff
@@ -0,0 +1,55 @@
+package example
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Makpoc/gomigen/types"
+)
+
+type ctxKey string
+
+const (
+	startTimeKey ctxKey = "startTime"
```

Comment:
The correct form for these is:

```go
type ctxStartTimeKey struct{}
```

and in the code below:

```go
ctx = context.WithValue(ctx, ctxStartTimeKey{}, startTime)
```

This causes the least allocation and processing, though it is a bit more complicated to read.